client/components/search/index.jsx

```diff
@@ -203,6 +205,11 @@ module.exports = React.createClass( {
 		if ( event.key === 'Escape' ) {
 			this.closeSearch( event );
 		}
+
+		if ( this.props.blurOnEnter && event.which === 13 ) {
```

Comment:
Is this behavior only desirable on `themes-search-card.jsx`? Wouldn't we want the soft keyboard to blur in any search situation when you hit enter? I would think we would want to somehow gate this on some indication that you have a virtual keyboard (ideal) or are on mobile (ok). I don't see any good approaches for detecting a virtual keyboard. We could maybe build something that infers this based on the viewport size changing when you focus an input element. Or what would you think of using [viewport.isMobile()](https://github.com/Automattic/wp-calypso/tree/master/client/lib/viewport) here instead of introducing a prop? Or maybe we could instead base this decision on how much of the screen is visible below the search input?